syntax = "proto3";
package game;

//心跳
message heart {
    string uuid = 1;  //玩家UUID
    string ext = 2;  //扩展字段
}

//连接服务器
message connect {
    string userId = 1;  //玩家唯一ID
    string deviceId = 2;  //玩家设备ID
}

//退出服务器
message exit {
    string userId = 1;  //玩家唯一ID
}

//加入房间
message joinRoom {
    string roomId = 1;  //房间ID
    string userId = 2;  //玩家唯一ID
}

//退出房间
message exitRoom {
    string roomId = 1;  //房间ID
    string userId = 2;  //玩家唯一ID
}

//解散房间
message dismissRoom {
    string roomId = 1;  //房间ID
}

//准备
message ready {
    string roomId = 1;  //玩家唯一ID
}

//投票
message vote {
    string userId = 1;  //玩家唯一ID
    string otherId = 2;  //被投票的玩家唯一ID
}

//发言
message speak {
    string userId = 1;  //玩家唯一ID
    string msg = 2;  //说话内容
    uint32 type = 3;  //类型 0：语音  1：文本
}

//释放技能
message skill {
    string userId = 1;  //玩家唯一ID
    uint32 sid = 2;  //技能ID
    string targetId = 3;  //技能施放对象ID
}